### PR TITLE
IDN expansion

### PR DESCRIPTION
**Pull request by philpennock**
* Grab IDN information from IANA
* Updated metadata for many many zones
* Add a human-targetted "ASCII mandated by policy" tag, use for `.vegas`, `.ai`
* Have an "IDN where ASCII blocked" domain, `.рф`
* Add tooling for showing the codepoint ranges (`-show-unicode`)

@ydnar @case

### CODE REVIEW

**Comment by case** [comment]
@ydnar any further comments and/or work needed on this?

**Comment by ydnar** [comment]
@case this currently has the `prohibitIDN` flag in metadata. Per thread elsewhere, this is ZoneDB state, not end-user state.

- Do we cache IDN table files?
- Where do we store hints about a given TLD like this?

**Comment by ydnar** [comment]
Looks like `-generate-go` isn’t idempotent, which could lead to large unnecessary diffs. Verified using the following:

```bash
go run build/cmd/zonedb/main.go -generate-go && git diff | head -n 50 > a.txt
go run build/cmd/zonedb/main.go -generate-go && git diff | head -n 50 > b.txt
go run build/cmd/zonedb/main.go -generate-go && git diff | head -n 50 > c.txt
go run build/cmd/zonedb/main.go -generate-go && git diff | head -n 50 > d.txt
diff a.txt b.txt
diff b.txt c.txt
diff c.txt d.txt
```

**Comment by ydnar** [comment]
@philpennock want to take a look at my changes?

File: build/cmd/zonedb/main.go [hunk 1]
```diff
@@ -176,7 +180,19 @@ func main() {
 		}
 	}
 
-	if *updateIDNTables || *updateAll {
+	// Explicitly _not_ under updateAll at this time, until we're confident this
+	// won't get us blacklisted on IANA's web-servers
+	if *updateIDNURLs {
```
Comment (ydnar) [inline build/cmd/zonedb/main.go]: Since this only fetches a single HTML file, shouldn’t this be safe under `*updateAll`?

**Comment by ydnar** [comment]
Per offline conversation regarding IDN, code point tables, and LGRs: I wanted to sketch out an alternate vision for this, where we constrain ZoneDB’s responsibility to the locations or descriptions of LGRs per zone, rather than an actual copy or implementation. Implementation could be left to a separate library, like `domainr/whois` or `domainr/epp`.

Hypothetical changes to the metadata JSON:

```json
{
  "...": "...",
  "labelGenerationRulesets": [
    {
      "script": "Latn",
      "url": "https://www.iana.org/domains/idn-tables/tables/com_latn_1.2.txt"
    },
    {
      "language": "ja",
      "url": "https://www.iana.org/domains/idn-tables/tables/com_ja_1.2.txt",
      "comment": "Japanese"
    }
  ]
}
```

**Comment by ydnar** [comment]
Closing this in favor of #674.

**Pull request closed without merging**